### ISSUE TITLE
Can't specify RequestPayer option in S3\ObjectCopier

### ISSUE THREAD

**Issue by AntoineWattier** [aws/aws-sdk-php#1232]
### Description

I came across an issue when trying to use the `ObjectCopier` with the `RequestPayer`option or any options in general.

### How to reproduce
 - one `source` bucket with the request payer option turned on and a random `file`on it
 - one `destination` bucket 

Try to copy one file from `source` to `destination` with the `ObjectCopier` class:

```php
$promise = new \Aws\S3\ObjectCopier(
    $s3Client,
    array(
        'Bucket' => 'source',
        'Key' => 'file',
    ),
    array(
        'Bucket' => 'destination',
        'Key' => 'file',
    ),
    'private',
    array(
        'RequestPayer' => 'requester',
    )
);

$promise->copy();
```
This will result in: 
```
Error executing "HeadObject" on "https://s3.amazonaws.com/source/file";
AWS HTTP error: Client error: `HEAD https://s3.amazonaws.com/source/file` resulted in a `403 Forbidden` response
```

Without the `RequestPayer` option the copy works perfectly.

### Why is it happening? 
The issue come from the fact that `ObjectCopier` doesn't specify options in the `HeadObject` call to retrieve the file metadata cf. https://github.com/aws/aws-sdk-php/blob/master/src/S3/ObjectCopier.php#L80

### Other related issue
Going further if we specify the options in the `HeadObject` call a simple copy will work, but not the Multipart one: 

```php
$promise = new \Aws\S3\ObjectCopier(
    $s3Client,
    array(
        'Bucket' => 'source',
        'Key' => 'largefile', // 10M file
    ),
    array(
        'Bucket' => 'destination',
        'Key' => 'largefile',
    ),
    'private',
    array(
         'RequestPayer' => 'requester',
         'mup_threshold' => 8388608 // 8M threshold
    )
);

$promise->copy();
```
This one will result in:
```
ERROR: An exception occurred while uploading parts to a multipart upload. The following parts had errors:
- Part 1: Error executing "UploadPartCopy" on "https://s3.amazonaws.com/destination/largefile?partNumber=1&uploadId=foo"; 
AWS HTTP error: Client error: `PUT https://s3.amazonaws.com/destination/largefile?partNumber=1&uploadId=foo` resulted in a `403 Forbidden` response
```
One more time options are not included in the commands (`UploadPartCopy` this time)
cf. https://github.com/aws/aws-sdk-php/blob/master/src/S3/MultipartCopy.php#L98

Let me know if I can provide more informations/traces to help you resolve this.

**Comment by imshashank** [comment]
Hey,

I tried the following code with no errors. I also see the object was copied to the new bucket.

Is there something I am missing?

Were you able to reproduce the issue with any other options being passed?

P.S. I tried this was an administrator account and hence have permission for everything.

```
<?php

require 'vendor/autoload.php';
use Aws\S3\S3Client;

$s3Client = new S3Client([
    'region' => 'us-west-2',
    'version' => '2006-03-01'
]);

$promise = new \Aws\S3\ObjectCopier(
    $s3Client,
    array(
        'Bucket' => 'bucket',
        'Key' => 'file',
    ),
    array(
        'Bucket' => 'bucket',
        'Key' => 'file',
    ),
    'private',
    array(
        'RequestPayer' => 'requester',
    )
);

$promise->copy();
```

**Comment by AntoineWattier** [comment]
Hi, 

I tried the transfer between 2 buckets of the same account and effectively there is no issue: I suppose the fact to be bucket owner overrides the `RequestPayer` header. 

But when you try with a distant bucket (with an user authorized in your `non-owned-bucket` policy) any `RequestPayer` request results in a 403.

Updated example: 
```php
<?php

require 'vendor/autoload.php';
use Aws\S3\S3Client;

$s3Client = new S3Client([
    'region' => 'us-west-2',
    'version' => '2006-03-01'
]);

$promise = new \Aws\S3\ObjectCopier(
    $s3Client,
    array(
        'Bucket' => 'non-owned-bucket',
        'Key' => 'file',
    ),
    array(
        'Bucket' => 'bucket',
        'Key' => 'file',
    ),
    'private',
    array(
        'RequestPayer' => 'requester',
    )
);

$promise->copy();

```

My `no-owned-bucket` policy is the following:
```json
{
	"Version": "2012-10-17",
	"Statement": [
		{
			"Sid": "AllowAll",
			"Effect": "Allow",
			"Principal": {
				"AWS": "arn:aws:iam::XXX"
			},
			"Action": [
				"s3:GetObject",
				"s3:ListBucket"
			],
			"Resource": [
				"arn:aws:s3:::non-owned-bucket",
				"arn:aws:s3:::non-owned-bucket/*"
			]
		}
	]
}```

**Comment by kstich** [comment]
@AntoineWattier Thanks again for raising this issue with us. Please see #1313 for the fix for this additional functionality for the ObjectUploader/MultipartCopy abstractions.

**Comment by kstich** [comment]
@AntoineWattier These options should now be able to be passed through via `$options['params']` as #1313 was merged in to [3.31.0](https://github.com/aws/aws-sdk-php/releases/tag/3.31.0).  This also introduces a 'before_lookup' callback for selective changes on the `HeadObject` command. Thanks again!
